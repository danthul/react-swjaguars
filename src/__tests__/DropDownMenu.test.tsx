import * as React from "react";
// import * as ReactDOM from "react-dom";

import { configure, shallow } from "enzyme";
import DropDownMenu, { DropDownMenuProps } from "../DropDownMenu";
import toJson from "enzyme-to-json";

// setup file
import * as Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

const noShowProps: DropDownMenuProps = {
  showDropDown: false
};

const showProps: DropDownMenuProps = {
  showDropDown: true
};

describe("Drop Down Menu", () => {
  // All tests will go here

  it("should render without menu", () => {
    const output = shallow(<DropDownMenu {...noShowProps} />);
    expect(toJson(output)).toMatchSnapshot();
  });

  it("should render with menu", () => {
    const output = shallow(<DropDownMenu {...showProps} />);
    expect(toJson(output)).toMatchSnapshot();
  });
});
