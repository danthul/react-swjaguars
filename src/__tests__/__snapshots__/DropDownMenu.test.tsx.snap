// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drop Down Menu should render with menu 1`] = `
<styled.ul
  showDropDown={true}
>
  <li>
    <Styled(Link)
      to="/"
    >
      Home
    </Styled(Link)>
  </li>
  <li>
    <Styled(Link)
      to="/calendar"
    >
      Calendar
    </Styled(Link)>
  </li>
  <li>
    <Styled(Link)
      to="/articles"
    >
      Messages
    </Styled(Link)>
  </li>
  <li>
    <Styled(Link)
      to="/slideshow"
    >
      Pictures
    </Styled(Link)>
  </li>
  <li>
    <Styled(Link)
      to="/about"
    >
      Our Mission
    </Styled(Link)>
  </li>
</styled.ul>
`;

exports[`Drop Down Menu should render without menu 1`] = `
<styled.ul
  showDropDown={false}
>
  <li>
    <Styled(Link)
      to="/"
    >
      Home
    </Styled(Link)>
  </li>
  <li>
    <Styled(Link)
      to="/calendar"
    >
      Calendar
    </Styled(Link)>
  </li>
  <li>
    <Styled(Link)
      to="/articles"
    >
      Messages
    </Styled(Link)>
  </li>
  <li>
    <Styled(Link)
      to="/slideshow"
    >
      Pictures
    </Styled(Link)>
  </li>
  <li>
    <Styled(Link)
      to="/about"
    >
      Our Mission
    </Styled(Link)>
  </li>
</styled.ul>
`;
